From cf86276055c05033ac6189c497c2b2ab2e32a5eb Mon Sep 17 00:00:00 2001
From: John Else <john.else@gmail.com>
Date: Mon, 6 Apr 2020 17:29:38 +0100
Subject: [PATCH] Make dhclient send our hostname to the DHCP server

This generates a config file of the form:

```
interface "xenbr0" {
  send host-name = gethostname();
  request subnet-mask, broadcast-address, time-offset, host-name, nis-domain, nis-servers, ntp-servers, interface-mtu, routers, domain-name, domain-name-servers;
}
```

Also fixes test build.

Signed-off-by: John Else <john.else@gmail.com>
---
 lib/network_utils.ml                 | 4 +++-
 test/network_test_lacp_properties.ml | 2 +-
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/lib/network_utils.ml b/lib/network_utils.ml
index 4e1c6d78..9b275ad5 100644
--- a/lib/network_utils.ml
+++ b/lib/network_utils.ml
@@ -746,6 +746,7 @@ struct
     Filename.concat "/var/lib/xcp" (Printf.sprintf "dhclient%s-%s.conf" ipv6' interface)
 
   let[@warning "-27"] generate_conf ?(ipv6=false) interface options =
+    let send = "host-name = gethostname()" in
     let minimal = ["subnet-mask"; "broadcast-address"; "time-offset"; "host-name"; "nis-domain";
                    "nis-servers"; "ntp-servers"; "interface-mtu"] in
     let set_gateway =
@@ -759,7 +760,8 @@ struct
       else (debug "%s is NOT the DNS interface" interface; [])
     in
     let request = minimal @ set_gateway @ set_dns in
-    Printf.sprintf "interface \"%s\" {\n  request %s;\n}\n" interface (String.concat ", " request)
+    Printf.sprintf "interface \"%s\" {\n  send %s;\n  request %s;\n}\n"
+      interface send (String.concat ", " request)
 
   let read_conf_file ?(ipv6=false) interface =
     let file = conf_file ~ipv6 interface in
diff --git a/test/network_test_lacp_properties.ml b/test/network_test_lacp_properties.ml
index 4e20de8c..ab53998b 100644
--- a/test/network_test_lacp_properties.ml
+++ b/test/network_test_lacp_properties.ml
@@ -50,7 +50,7 @@ let test_lacp_aggregation_key arg () =
 module OVS_Cli_test = struct
   include Ovs.Cli
   let vsctl_output = ref []
-  let vsctl ?log args =
+  let vsctl ?log:_ args =
     vsctl_output := args ;
     String.concat " " args
 end
